controller/api/public/client.go

```diff
@@ -19,44 +19,52 @@ import (
 )
 
 const (
-	ApiRoot             = "/" // Must be absolute (with a leading slash).
-	ApiVersion          = "v1"
-	JsonContentType     = "application/json"
-	ApiPrefix           = "api/" + ApiVersion + "/" // Must be relative (without a leading slash).
-	ProtobufContentType = "application/octet-stream"
-	ErrorHeader         = "conduit-error"
+	ApiRoot                 = "/" // Must be absolute (with a leading slash).
+	ApiVersion              = "v1"
+	JsonContentType         = "application/json"
+	ApiPrefix               = "api/" + ApiVersion + "/" // Must be relative (without a leading slash).
+	ProtobufContentType     = "application/octet-stream"
+	ErrorHeader             = "conduit-error"
+	ConduitApiSubsystemName = "conduit-api"
 )
 
-type client struct {
-	serverURL *url.URL
-	client    *http.Client
+type grpcOverHttpClient struct {
+	serverURL  *url.URL
+	httpClient *http.Client
 }
 
 type tapClient struct {
 	ctx    context.Context
 	reader *bufio.Reader
 }
 
-func (c *client) Stat(ctx context.Context, req *pb.MetricRequest, _ ...grpc.CallOption) (*pb.MetricResponse, error) {
+func (c *grpcOverHttpClient) Stat(ctx context.Context, req *pb.MetricRequest, _ ...grpc.CallOption) (*pb.MetricResponse, error) {
 	var msg pb.MetricResponse
 	err := c.apiRequest(ctx, "Stat", req, &msg)
 	return &msg, err
 }
 
-func (c *client) Version(ctx context.Context, req *pb.Empty, _ ...grpc.CallOption) (*pb.VersionInfo, error) {
+func (c *grpcOverHttpClient) Version(ctx context.Context, req *pb.Empty, _ ...grpc.CallOption) (*pb.VersionInfo, error) {
 	var msg pb.VersionInfo
 	err := c.apiRequest(ctx, "Version", req, &msg)
 	return &msg, err
 }
 
-func (c *client) ListPods(ctx context.Context, req *pb.Empty, _ ...grpc.CallOption) (*pb.ListPodsResponse, error) {
+func (c *grpcOverHttpClient) SelfCheck(ctx context.Context, req *common.SelfCheckRequest, _ ...grpc.CallOption) (*common.SelfCheckResponse, error) {
+	var msg common.SelfCheckResponse
+	err := c.apiRequest(ctx, "SelfCheck", req, &msg)
+	return &msg, err
+}
+
+func (c *grpcOverHttpClient) ListPods(ctx context.Context, req *pb.Empty, _ ...grpc.CallOption) (*pb.ListPodsResponse, error) {
 	var msg pb.ListPodsResponse
 	err := c.apiRequest(ctx, "ListPods", req, &msg)
 	return &msg, err
 }
 
-func (c *client) Tap(ctx context.Context, req *pb.TapRequest, _ ...grpc.CallOption) (pb.Api_TapClient, error) {
-	rsp, err := c.post(ctx, "Tap", req)
+func (c *grpcOverHttpClient) Tap(ctx context.Context, req *pb.TapRequest, _ ...grpc.CallOption) (pb.Api_TapClient, error) {
+	url := c.endpointNameToPublicApiUrl("Tap")
+	rsp, err := c.post(ctx, url, req)
```

Comment:
For my own understanding, can you explain this refactor a bit more? Why pull functionality out of the `post` method into a separate `endpointNameToPublicApiUrl` method, if you're only ever passing the output of that method back into the `post` method?

Reply:
Ahhh, right -- thanks. I totally missed that log message.